vm/src/vm/runners/cairo_runner.rs

```diff
@@ -3217,6 +3243,9 @@ mod tests {
     }
 
     /// Test that `get_output()` works when the `output` builtin is not the first one.
+    // TODO review: this test swaps builtins, but builtins should appear in the order
+    //              specified in the program (i.e. ordered).
+    #[ignore = "this test does not seem to make sense"]
```

Comment:
Is this test now failing???
If it fails can we correct it?

Reply:
Yes, this test is failing. I'd need to disassemble it or find the source to be sure, but it feels like this test was built ad hoc for something that should not happen (the builtins stack being out of order). If you look at the previous stack initialization code, it looks like this:

```
        for builtin_runner in vm.builtin_runners.iter() {
            stack.append(&mut builtin_runner.initial_stack());
        }
```

In the test, they swap the builtins so the output is `[<bitwise_base>, <output_base>]`. The only way this can work is if the program is aware of that and reverts the order of builtins. IMO this should never happen, hence why I disabled the test and suggested its removal. But I might be missing something. I think @azteca1998 wrote this test, maybe he could explain what it does in more details?

Reply:
A different builtin order is used for starknet contracts, maybe we could use [initialize_all_builtins](https://github.com/lambdaclass/cairo-vm/blob/6ae7b32b0cb8cf758e52fbf795d6689b81a7ca0e/vm/src/vm/runners/cairo_runner.rs#L350) in the test  instead of `initialize_builtins` (and remove output from the program builtins) to test a proper use case where ouput is not the first builtin

Reply:
This test was added because the write_output function used to asume that the output builtin was always the first one, which is not always the case in a starknet context. 
For the starknet context we use [initialize_function_runner](https://github.com/lambdaclass/cairo-vm/blob/6ae7b32b0cb8cf758e52fbf795d6689b81a7ca0e/vm/src/vm/runners/cairo_runner.rs#L1217), that initializes first the builtins in the order that they appear in the program, but then also initializes the remaining builtins following starknet's builtins order, which is different from the vm's expected builtin order

Reply:
After testing a few solutions, here's what's possible.

1. This PR makes it impossible to run a standard program with builtins in the wrong order. Builtins can be specified in any order on the VM struct. We check that the builtins are specified in the standard order in the program and return a `DisorderedBuiltins` error if the order is not respected. The swap of builtins on the VM is now meaningless.

2. As you mentioned `initialize_function_runner`, running it instead of `initialize_builtins` works. I chose this solution for the moment.